src/functional/ffront/foast_passes/utils.py

```diff
@@ -0,0 +1,47 @@
+from functional.ffront import field_operator_ast as foast
+
+
+def compute_assign_indices(
+    targets: list[foast.FieldSymbol | foast.TupleSymbol | foast.ScalarSymbol | foast.Starred],
+    num_elts: int,
+) -> list[tuple[int, int] | int]:
+    """
+    Compute a list of relative indices, mapping each target to its respective value(s).
+
+    This function is used when mapping a tuple of targets to a tuple of values, and also handles the
```

Comment:
```suggestion
    This function is used when assigning a tuple of values to a tuple target, and also handles the
```
Nitpicking, but the word `mapping` and the ordering is rather confusing.